packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol

```diff
@@ -55,87 +91,94 @@ contract ZoraCreator1155PremintExecutorImpl is Ownable2StepUpgradeable, UUPSUpgr
+    /// or match the contract admin on the contract creation config if the contract hasn't been created yet.
+    /// Contract address of the created contract is deterministically generated from the contract config and this contract's address.
+    /// @param contractConfig Parameters for creating a new contract, if one doesn't exist yet.  Used to resolve the deterministic contract address.
+    /// @param premintConfig Parameters for creating the token, and minting the initial x tokens to the executor.
+    /// @param signature Signature of the creator of the token, which must match the signer of the premint config, or have permission to create new tokens on the erc1155 contract if it's already been created
+    /// @param quantityToMint How many tokens to mint to the executor of this transaction once the token is created
+    /// @param mintArguments Abi encoded additional mint arguments: including mintComment and mintReferral
+    function premint(
+        ContractCreationConfig calldata contractConfig,
+        PremintConfig calldata premintConfig,
+        bytes calldata signature,
+        uint256 quantityToMint,
+        bytes memory mintArguments
+    ) public payable returns (uint256 newTokenId) {
+        // get or create the contract with the given params
+        // contract address is deterministic.
+        (IZoraCreator1155 tokenContract, bool isNewContract) = ZoraCreator1155PremintExecutorImplLib.getOrCreateContract(zora1155Factory, contractConfig);
 
-        if (isNewContract) {
-            // if address doesn't exist for hash, create it
-            tokenContract = _createContract(contractConfig);
-        } else {
-            tokenContract = IZoraCreator1155(contractAddress);
-        }
-    }
+        // assume contract has legacy interface expecting v1 signatures; call it.
+        newTokenId = IZoraCreator1155PremintV1Signatures(address(tokenContract)).delegateSetupNewToken(premintConfig, signature, msg.sender);
 
-    function _createContract(ContractCreationConfig calldata contractConfig) private returns (IZoraCreator1155 tokenContract) {
-        // we need to build the setup actions, that must:
-        bytes[] memory setupActions = new bytes[](0);
-
-        // create the contract via the factory.
-        address newContractAddresss = zora1155Factory.createContractDeterministic(
-            contractConfig.contractURI,
-            contractConfig.contractName,
-            // default royalty config is empty, since we set it on a token level
-            ICreatorRoyaltiesControl.RoyaltyConfiguration({royaltyBPS: 0, royaltyRecipient: address(0), royaltyMintSchedule: 0}),
-            payable(contractConfig.contractAdmin),
-            setupActions
+        ZoraCreator1155PremintExecutorImplLib.performMint(tokenContract, premintConfig.tokenConfig.fixedPriceMinter, newTokenId, quantityToMint, mintArguments);
+
+        // emit legacy Preminted event
+        emit Preminted(
```

Comment:
can we emit v2 of Preminted event? maybe we can not include full token config as to not break compatibility